frontend/src/components/Table/Table.tsx

```diff
@@ -188,10 +201,15 @@ const IndeterminateCheckbox = React.forwardRef(
       if (isHeaderCheck) {
         setSelected([]);
       } else {
-        if (currentSelected.find(selected => selected.id === row.original.id)) {
-          remove(currentSelected, row.original);
-        } else {
+        if (isSingleSelect === true) {
```

Comment:
If isSingleSelect is passed to the table, only one row can be selected at a time.